app/controllers/download_controller.rb

```diff
@@ -14,8 +14,8 @@ class DownloadController < ApplicationController
           download_type: "generated-#{params[:type]}"
         })
     respond_to do |format|
-      format.json { render json: flash, response: response }
-      format.html { render json: flash, response: response }
+      format.json { render json: [flash, ""], response: response, status: 500 }
+      format.html { render json: [flash, ""], response: response, status: 500 }
```

Comment:
return a 500 response when there is an error, so our new downloads controller can correctly detect via status

also make the response from an exception match the response from a success (which changed to include the filename when available)